syntax = "proto3";

service RobotService {
  rpc GetRobots(Empty) returns (stream RobotData);
  rpc RegisterRobot(RobotInfo) returns (RobotData);
  rpc UnregisterRobot(RobotData) returns (Status);
  rpc CheckRobot(RobotData) returns (Status);
  rpc GetCaptain(Empty) returns (RobotData);
  rpc HealthCheck(RobotData) returns (Commands);
  rpc RegisterCaptain(RobotData) returns (Status);
}

message RobotInfo {
  string name = 1;
}

message RobotData {
  int32 id = 1;
  string name = 2;
}

message Status {
  bool success = 1;
}

message Empty {}

message Commands {
	bool connected = 1;
	bool elect = 2;
}